cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -680,43 +680,100 @@ public boolean apply(RunRecordMeta runRecordMeta) {
   }
 
   public void upgradeVersionKeys() {
-    MDSKey startKey = new MDSKey.Builder().add(TYPE_APP_META).build();
-    Map<MDSKey, ApplicationMeta> appMetas = listKV(startKey, ApplicationMeta.class);
-    Map<MDSKey, ApplicationMeta> newAppMetas = new HashMap<>();
-    for (Map.Entry<MDSKey, ApplicationMeta> oldKey : appMetas.entrySet()) {
-      MDSKey newKey = appendDefaultVersion(oldKey.getKey());
-      newAppMetas.put(newKey, oldKey.getValue());
+    upgradeVersionKeys(TYPE_APP_META, ApplicationMeta.class);
+    upgradeVersionKeys(TYPE_RUN_RECORD_STARTED, RunRecordMeta.class);
+    upgradeVersionKeys(TYPE_RUN_RECORD_SUSPENDED, RunRecordMeta.class);
+    upgradeVersionKeys(TYPE_RUN_RECORD_COMPLETED, RunRecordMeta.class);
+    upgradeVersionKeys(TYPE_WORKFLOW_NODE_STATE, RunRecordMeta.class);
```

Comment:
Yes, because all the queries in `AppMetadataStore` calls `getProgramKeyBuilder` to construct MDSKey, which always includes app version